js/rank_states_utils.js

```diff
@@ -30,6 +35,18 @@ function rankEm() {
 	
 	var bardata = arguments[1];
 	
+	var x = d3.scaleLinear()
+	  .range([0, rankSvg.width])
+	  .domain([0, bardata.length]);
+	
+	var barWidth = rankSvg.width / bardata.length * 0.92;
```

Comment:
We can make even fuller use of D3 for the x scale by using `d3.scaleBand` instead of `d3.scaleLinear`. See https://bl.ocks.org/d3indepth/1aef77d17863e603ff4e84226db5b227 for an example. `scaleBand` allows you to set the padding and total range and get back `bandScale.bandwidth()` (i.e., it computes barWidth for you). And it deals with any x nudging due to padding without you having to set any numbers. The benefits of scaleBand over scaleLinear are pretty minor, but might be worth switching just to practice the D3 way.